addons/project-folders/userscript.js

```diff
@@ -0,0 +1,191 @@
+export default async function ({ addon, global, console, msg }) {
+  let projectColumns;
+  projectColumns = await addon.tab.waitForElement("ul.media-list", {
+    markAsSeen: true,
+  });
+
+  let foldersJSON;
+  const projectTab = document.querySelectorAll("li.first")[0];
+  projectTab.childNodes[0].onclick = async () => {
+    projectColumns = await addon.tab.waitForElement("ul.media-list", {
+      markAsSeen: true,
+    });
+    createFolderAreaAndButton();
+    foldersJSON = await load();
+  };
+
+  if (document.querySelectorAll("div.folders-container")[0] === undefined) {
+    createFolderAreaAndButton();
+    foldersJSON = await load();
+  }
+
+  async function getProjectDetails(projectID) {
+    const token = await addon.auth.fetchXToken();
+    const projectDetails = await (
+      await fetch(`https://api.scratch.mit.edu/projects/${projectID}/`, {
+        headers: {
+          "content-type": "application/json",
+          "x-csrftoken": addon.auth.crsfToken,
+          "x-token": token,
+        },
+      })
+    ).json();
+
+    return projectDetails;
+  }
+
+  async function load() {
+    const foldersJSON = { folders: [] };
+
+    const folderDiv = document.querySelectorAll(".folders")[0];
+
+    const noFolderSpan = document.createElement("span");
+    noFolderSpan.textContent = msg("loading");
+    folderDiv.appendChild(noFolderSpan);
+
+    const tempBR = document.createElement("br");
+    folderDiv.appendChild(tempBR);
+
+    const loader = document.createElement("progress");
+    loader.setAttribute("value", "0");
+    loader.setAttribute("max", "100");
+    folderDiv.appendChild(loader);
```

Comment:
Does this load slow enough to require a progress bar? If so it should probably be made more performant

Reply:
I added a progress bar to tell the use the amount of time it is taking. But if the user has less projects then it will take less time but I don't think the performance can be made any better.